src/core_plugins/timelion/server/lib/offset_time.js

```diff
@@ -34,3 +34,29 @@ export default function offsetTime(milliseconds, offset, reverse) {
   const momentObj = moment(milliseconds)[mode](parts[1], parts[2]);
   return momentObj.valueOf();
 }
+
+/*
+ * Calculate offset when parameter is requesting a relative offset based on requested time range.
+ *
+ * @param {string} offset - offset parameter value
+ * @param {number} from - kibana global time 'from' in milliseconds
+ * @param {number} to - kibana global time 'to' in milliseconds
+ */
+export function preprocessOffset(offset, from, to) {
+  if (!offset.startsWith('kibana_time')) {
+    return offset;
+  }
+
+  const parts = offset.split(':');
+  let factor = -1;
```

Comment:
I know that more people might want to use this for negative shifting. I still have the feeling `1` would be the more intuitive default if you don't specify anything.

Maybe @alexfrancoeur and @ppisljar could also give his idea about that: if you see a timelion expression like `.es(offset=timerange)` would you expect it to be shifted once the chart time range to the past or future, i.e. should it be equal to `.es(offset=timerange:1)` or `.es(offset=timerange:-1)`?

Reply:
i guess -1, however none of them are really intuitive to me, i would actually vote for no default.